### DISCUSSION TITLE
Keep named plugins

### DISCUSSION THREAD

**Discussion by sudiptochoudhury** [GrapesJS/grapesjs#5390] [Roadmap Suggestions]
Hi,

I would like to request you to keep named plugins feature. Currently I see it's been deprecated.

Use cases: 

If the plugins have names (ids) it's easy to generate the editor config's `plugins`  array as array of string from backend.
I have an editor whose config is dynamically generated based on various criteria. Based on different criteria and access credentials I may or may not allow some plugins to be added to the editor. 

I also need to load a based plugin as the very first loaded plugin and many other subsequent plugins are dependent on the functionality added by this base plugin. While using a string name, it's easy now to sort the plugins array before passing it to editor init.

I understand that, I can probably develop an additional plugin store with ids and associate each with the plugin function. But, given that the feature already exists (now being deprecated) in grapesjs, I would like to request you to keep it, unless of course you tell me that keeping the feature is limiting Grapesjs in various ways.

**Comment by artf** [comment]
The point of deprecating the `PluginManager.add` method was to discourage people from using it to create new external plugins but there are no plans to remove the `PluginManager` itself (that still can be used to add/create named plugins in your application domain) so you should be safe.

**Comment by sudiptochoudhury** [comment]
Thanks.
